contracts/RewardPool.sol

```diff
@@ -92,13 +92,39 @@ contract RewardPool is Initializable, Versionable, Ownable {
     return true;
   }
 
+  function valid(
+    bytes calldata leaf,
+    bytes calldata proof
+  ) external view returns (bool) {
+    uint256 paymentCycleNumber = uint256(leaf[4]);
+    uint256 root = payeeRoots[paymentCycleNumber];
+    return proof.verify(root, keccak256(leaf));
+  }
+
+  function claimed(leaf) external view returns (bool) {
+    uint256 paymentCycleNumber = uint256(leaf[4]);
+    address rewardProgramID = address(leaf[0]);
+    address payableToken = address(leaf[1]);
+    address rewardee = address(leaf[2]);
+    return rewardsClaimed[paymentCycleNumber][rewardProgramID][payableToken][rewardee];
+  }
+
   function claim(
-    address rewardProgramID,
-    address payableToken,
-    uint256 amount,
+    bytes calldata leaf,
     bytes calldata proof
   ) external returns (bool) {
-    require(amount > 0, "Cannot claim non-positive amount");
+
+    uint256 rewardProgramID = uint256(leaf[0]);
```

Comment:
why is rewardProgramID a `uint256`. I previously used the id as an ethereum address.